### ISSUE TITLE
XHR error (404 Not Found) loading crypto.js when running gulp

### ISSUE THREAD

**Issue by andrewkdouglas** [madhukard/angular2-jspm-seed#11]
Hi,

I cloned you repo, ran npm and jspm install, then gulp.

All these run fine, but then running gulp causes the following console output in chrome:

```
angular2-polyfills.min.js:1 Error: Error: XHR error (404 Not Found) loading http://localhost:3000/crypto.js(…)
e.invoke @ angular2-polyfills.min.js:1
e.run @ angular2-polyfills.min.js:1
(anonymous function) @ angular2-polyfills.min.js:1
e.invokeTask @ angular2-polyfills.min.js:1
e.runTask @ angular2-polyfills.min.js:1
a @ angular2-polyfills.min.js:1
invoke @ angular2-polyfills.min.js:1
```

**Comment by hamjak** [comment]
+1

**Comment by ksteigerwald** [comment]
jspm install crypto

**Comment by madhukard** [comment]
Fixed

**Comment by rn1d** [comment]
But why is `crypto` needed with Angular 2 + JSPM - can anyone comment? 

I had the same issue when upgrading to RC.1. `jspm install crypto` solved the issue for me, too. It seems that Angular 2 runs just fine without it when using just NPM. Also, Angular ran without installing crypto on Beta 16, and the problem came up again with RC.1.

My apologies if commenting on a closed issue is bad form.

**Comment by RohitLuthra19** [comment]
I am also facing this problem again and again its showing an error 
I have also used npm install crypto ...
But issue not solved

![image](https://cloud.githubusercontent.com/assets/11659536/15206985/9b6ea10a-1840-11e6-80ba-9b34d714abd1.png)

**Comment by harryjubb** [comment]
@rn1d: Good question! I would like to know too, if anyone can help. For curiosity more than anything, at least it works!

@RohitLuthra19 

Are you using JSPM?

If so ensure:

```
jspm install crypto
```

not

```
npm install crypto
```

**Comment by rn1d** [comment]
@harryjubb I think I've got an idea of what could have been going on. I'm not sure I'm right, but I hope the following gives you some clues. 

If you look at the JSPM registry (https://github.com/jspm/registry/tree/master/package-overrides/npm), you'll see that `reflect-metadata` has an override for `crypto` (https://github.com/jspm/registry/blob/master/package-overrides/npm/reflect-metadata%400.1.3.json) - I don't know what exactly this does, but it seems to me that it says `crypto` should be ignored for browsers. Also, there're no overrides for version 1.2, which, if I remember correctly, was used by Angular 2 at one point. I think this is what was causing the issue.

What I ended up doing - I don't rely on JSPM/System.js to load shims (as the Angular team suggests). Instead I load bundled shims (reflect-metadata, es6-shim, zone.js) built from npm packages with a `<script>` tag.
